pkgs/development/libraries/science/math/petsc/default.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenv, 
+fetchgit, 
+mpi,
+openblas,
+lapack ? null,
+blasName ? "",
+lapackName ? "",
+pkgconfig,
+python 
+}:
+
+# use openblas for both the blas and lapack backend by default
+# provide necessary option if the user want to override to use Intel MKL or other
+
+let
+    blas = openblas;
+    liblapack = if (lapack == null)  then openblas else lapack;
+    blasLibName = if (blasName != "") then blasName else "openblas";
+    liblapackLibName = if(lapackName != "") then lapackName else "openblas";
+	defaultOptFlags = "-g -O2 -fstack-protector";
```

Comment:
I believe stack protector flags will be added by default now. Is that not sufficient?